src/code/sys_matrix.c

```diff
@@ -1,89 +1,1945 @@
+            temp2 *= temp2;
+            temp2 += SQ(temp3);
+            temp3 = src->zz;
+            temp2 += SQ(temp3);
+            temp2 = sqrtf(temp2);
+            temp2 = temp3 / temp2;
+
+            dest->x = Math_Atan2S(temp, temp2);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_matrix/Matrix_InsertRotationAroundUnitVector_f.s")
+/**
+ * @brief Rotate by `angle` radians about `axis`, which is assumed to be a unit vector.
+ *
+ * @param angle rotation angle (radians).
+ * @param axis axis about which to rotate, must be a unit vector.
+ * @param mode APPLY or NEW.
+ *
+ * @remark original name may have been "Matrix_RotateVector", but clashed with the next function.
+ */
+void Matrix_RotateAxisF(f32 angle, Vec3f* axis, MatrixMode mode) {
+    MtxF* cmf;
+    f32 sin;
+    f32 cos;
+    f32 versin;
+    f32 temp1;
+    f32 temp2;
+    f32 temp3;
+    f32 temp4;
+    f32 temp5;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_matrix/Matrix_InsertRotationAroundUnitVector_s.s")
+    if (mode == MTXMODE_APPLY) {
+        if (angle != 0) {
+            cmf = sCurrentMatrix;
+
+            sin = sinf(angle);
+            cos = cosf(angle);
+
+            temp1 = cmf->xx;
+            temp2 = cmf->xy;
+            temp3 = cmf->xz;
+            temp4 = (axis->x * temp1 + axis->y * temp2 + axis->z * temp3) * (1.0f - cos);
+            cmf->xx = temp1 * cos + axis->x * temp4 + sin * (temp2 * axis->z - temp3 * axis->y);
+            cmf->xy = temp2 * cos + axis->y * temp4 + sin * (temp3 * axis->x - temp1 * axis->z);
+            cmf->xz = temp3 * cos + axis->z * temp4 + sin * (temp1 * axis->y - temp2 * axis->x);
+
+            temp1 = cmf->yx;
+            temp2 = cmf->yy;
+            temp3 = cmf->yz;
+            temp4 = (axis->x * temp1 + axis->y * temp2 + axis->z * temp3) * (1.0f - cos);
+            cmf->yx = temp1 * cos + axis->x * temp4 + sin * (temp2 * axis->z - temp3 * axis->y);
+            cmf->yy = temp2 * cos + axis->y * temp4 + sin * (temp3 * axis->x - temp1 * axis->z);
+            cmf->yz = temp3 * cos + axis->z * temp4 + sin * (temp1 * axis->y - temp2 * axis->x);
+
+            temp1 = cmf->zx;
+            temp2 = cmf->zy;
+            temp3 = cmf->zz;
+            temp4 = (axis->x * temp1 + axis->y * temp2 + axis->z * temp3) * (1.0f - cos);
+            cmf->zx = temp1 * cos + axis->x * temp4 + sin * (temp2 * axis->z - temp3 * axis->y);
+            cmf->zy = temp2 * cos + axis->y * temp4 + sin * (temp3 * axis->x - temp1 * axis->z);
+            cmf->zz = temp3 * cos + axis->z * temp4 + sin * (temp1 * axis->y - temp2 * axis->x);
+        }
+    } else {
+        cmf = sCurrentMatrix;
+
+        if (angle != 0) {
+            sin = sinf(angle);
+            cos = cosf(angle);
+            versin = 1.0f - cos;
+
+            cmf->xx = axis->x * axis->x * versin + cos;
+            cmf->yy = axis->y * axis->y * versin + cos;
+            cmf->zz = axis->z * axis->z * versin + cos;
+
+            if (0) {}
```

Comment:
```suggestion
            if (1) {}
```
Does it have to be 0?

Reply:
For some reason it does.